packages/react-server/src/ReactFizzServer.js

```diff
@@ -1779,6 +1809,9 @@ export function performWork(request: Request): void {
       // we'll to restore the context to what it was before returning.
       switchContext(prevContext);
     }
+    if (yieldedForPendingTasks) {
+      scheduleWork(() => performWork(request));
+    }
```

Comment:
Has to go here to avoid reentrancy. Could still be sync in some environments so it's only yieldy if the scheduler is async